Runtime/Core/ProBuilderMeshFunction.cs

```diff
@@ -368,6 +368,18 @@ public void ToMesh(MeshTopology preferredTopology = MeshTopology.Triangles)
 
             mesh.subMeshCount = submeshes.Length;
 
+            if (mesh.subMeshCount == 0)
+            {
+                EnsureMeshFilterIsAssigned();
+
+                if (usedInParticleSystem)
+                    MeshUtility.RestoreParticleSystem(this);
+
+                IncrementVersionIndex();
+
+                return;
+            }
```

Comment:
Would be nice if the code duplication could be avoided (i.e. compared to the very bottom code of this function). Also a comment might be nice for future changes that would explain why the materials/submesh code below can't be executed/is problematic when subMeshCount == 0.

Reply:
Good point, thanks for the feedback! Fixed in [4e34729](https://github.com/Unity-Technologies/com.unity.probuilder/pull/470/commits/4e34729e31f3f46b08484045802645828293c616)